### DISCUSSION TITLE
How to programatically change repo settings

### DISCUSSION THREAD

**Discussion by PythonCoderAS** [community/community#23459] [New to GitHub]
<p>I want to enable the “Automatically delete head branches” setting on all of my repos and forks, but I have over 50 repos, so this would take a lot of time. Is there a way to automate this via the GitHub API?</p>
<p>A related topic seems to be <a href="https://github.community/t/anyway-to-set-repo-features-settings/192963" class="inline-onebox">Anyway to set repo features/settings?</a> but it never got any replies.</p>

**Comment by mpboom** [comment, marked as answer]
<p>You’re probably looking for the <strong>Update a repository</strong> endpoint in the API:<br>
<a href="https://docs.github.com/en/rest/repos/repos#update-a-repository" class="onebox" target="_blank" rel="noopener nofollow ugc">https://docs.github.com/en/rest/repos/repos#update-a-repository</a></p>

**Comment by laughedelic** [comment]
There are a couple of existing solutions for settings management:
- [repository-settings/app](https://github.com/repository-settings/app) settings config at the repo level
- [github/safe-settings](https://github.com/github/safe-settings) centralized org settings
